src/create-changefile.ts

```diff
@@ -0,0 +1,115 @@
+            access_type: 'offline',
+            scope: SCOPES,
+        });
+        console.log('Authorize this app by visiting this url:', authUrl);
+        const rl = readline.createInterface({
+            input: process.stdin,
+            output: process.stdout,
+        });
+        await rl.question('Find code in URL at last redirect after code=<CODE>. Enter here: ', (code: any) => {
+            rl.close();
+            oAuth2Client.getToken(code, async (err: any, token: any) => {
+                if (err) return console.error('Error while trying to retrieve access token', err);
+                oAuth2Client.setCredentials(token);
+                // Store the token to disk for later program executions
+                fs.writeFile(TOKEN_PATH, JSON.stringify(token), (err: any) => {
+                    if (err) return console.error(err);
+                    console.log('Token stored to', TOKEN_PATH);
+                });
+            });
+        });
+    } else {
+        filename = `tx_${new Date().getTime()}.json`;
+        await createJsonOutput(oAuth2Client, sheetName, filename);
+    }
+
+    return filename;
+}
+
+async function createJsonOutput(auth: any, sheetName: string, filename: string): Promise<void> {
+    const sheets = google.sheets({ version: 'v4', auth });
+    let result;
+    try {
+        result = await sheets.spreadsheets.values.get({
+            spreadsheetId: '1PwKizspl84t7AG2QlQMlr77OxjcuEY_mq8JwTu9ptQM',
+            range: `${sheetName}!A2:J`,
+        });
+    } catch (e) {
+        throw Error('Could not find sheetname provided.');
+    }
+    const rows: string[] = result.data.values as unknown as string[];
+
+    if (rows.length) {
+        const masterchefContract = new Contract(
+            '0x8166994d9ebbe5829ec86bd81258149b87facfd3',
+            masterchefAbi,
+            new ethers.providers.JsonRpcProvider('https://rpc.ftm.tools'),
+        );
+
+        let outJson = [];
+
+        for (const row of rows) {
+            const isNew = row[0];
+            const poolName = row[1];
+            const masterchefPoolId = row[2];
+            const newAllocationPoints = row[7];
+            const lpToken = row[9];
+
+            if (masterchefPoolId) {
+                const poolInfo = await masterchefContract.poolInfo(row[2]);
+                const currentAllocationPoints = BigNumber.from(poolInfo.allocPoint).toString();
+
+                if (newAllocationPoints !== currentAllocationPoints) {
+                    outJson.push({
+                        type: 'edit',
+                        pid: parseFloat(masterchefPoolId),
+                        allocationPoints: parseFloat(newAllocationPoints),
+                    });
+                    console.log(
+                        `Need to change alloc for pool ${poolName} to ${newAllocationPoints} from ${currentAllocationPoints}`,
+                    );
+                }
+            }
+            if (isNew) {
```

Comment:
would rather do an `else if(isNew)` or even more defensive, like it cannot have  a `masterChefPoolId` and a `isNew` flat at the same time. so maybe even throw if both is present to prevent fuck ups